index.md

```diff
@@ -2,9 +2,286 @@
+
+#### Quotas
+
+Dataset file quotas for RSP users' POSIX file space will be monitored by the RSP infrastructure itself.
+Users will not have direct access to buckets without being mediated by the Butler server issuing signed URLs.
+
+For butler datasets quota management depends on how datasets are stored.
+For a shared bucket the butler server has to know (a) how much quota a user has, and (b) how much of the allocation they have utilized, before deciding whether a dataset can be stored.
+Care must be taken to deal with a user attempting multiple writes in parallel.
+
+Running a daily "cron" job to scan the bucket to calculate quota allocation does not seem feasible given how much people could store in a single day (especially for the outputs of user batch).
+This likely means that there will have to be some lightweight database tracking quotas for each user across all their buckets, with each write or deletion adjusting the quota value as appropriate.
+
+```{note}
+Since the system doesn't know the size of the file until it has been written with the signed URL, does that mean that the file can be deleted by the server and the write rejected if it causes the user to exceed their quota?
+Are they allowed to write one file over their quota?
+What about parallel writes?
+How strict are the requirements?
+```
+
+The per-user/group bucket makes everything significantly easier in that the storage system knows how much data is in each bucket without having to check the size of individual files.
+Querying the bucket usage (does Google have an API for this or is it solely visible on admin screens?) and the quota allocation is then sufficient to decide if a write can be accepted or not and there is no need to track individual deletions and writes.
+
+### Prompt Processing Repository
+
+The prompt processing repository differs from a data release repository in that it is being updated on a daily basis for the entire duration of the survey, but also datasets are regularly being deleted from it.
+User products stored in the repository will include provenance to datasets that only exist in registry and can not themselves be retrieved.
+
+The long life time of the repository indicates that there will have to be migrations to support the evolving LSST Science Pipelines and evolving butler registry schema migrations.
+
+For example, at the end of the survey there will be no datasets written by prompt processing in the repository but there might be derived user products.
+Users are unlikely to want their derived products to be deleted automatically if they have sufficient quota available.
+
+Reading these old files might be difficult given data model evolution, and possibilities for supporting this are discussed in the [next section](#accessing-multiple-data-releases).
+
+```{note}
+Will the DR10 data release pipelines software be required to read (without loss of information) datasets written by the DR1 software?
+```
```

Comment:
I think "read without loss of information" is viable.

Running later steps of the DR10 pipelines against the outputs of earlier steps of the DR1 pipeline will definitely not be supported.

"Read into an object with the same Python interface that it had in DR1" is also not something I think we should guarantee, but I think we can do it in most cases.